demos/mqtt/mqtt_demo_connection_sharing/mqtt_demo_connection_sharing.c

```diff
@@ -0,0 +1,1517 @@
+
+        /* Send MQTT CONNECT packet to broker. */
+        mqttStatus = MQTT_Connect( pMqttContext, &connectInfo, NULL, CONNACK_RECV_TIMEOUT_MS, pSessionPresent );
+
+        if( mqttStatus != MQTTSuccess )
+        {
+            returnStatus = EXIT_FAILURE;
+            LogError( ( "Connection with MQTT broker failed with status %u.", mqttStatus ) );
+        }
+        else
+        {
+            LogInfo( ( "MQTT connection successfully established with broker.\n\n" ) );
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int disconnectMqttSession( MQTTContext_t * pMqttContext )
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    int returnStatus = EXIT_SUCCESS;
+
+    assert( pMqttContext != NULL );
+
+    /* Send DISCONNECT. */
+    mqttStatus = MQTT_Disconnect( pMqttContext );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "MQTT disconnect failed: Error=%s",
+                    MQTT_Status_strerror( mqttStatus ) ) );
+        returnStatus = EXIT_FAILURE;
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static int subscribeToTopic( MQTTContext_t * pMqttContext,
+                             const char * pTopicFilter,
+                             uint16_t topicFilterLength )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus;
+    MQTTSubscribeInfo_t pSubscriptionList[ 1 ];
+
+    assert( pMqttContext != NULL );
+
+    /* Start with everything at 0. */
+    ( void ) memset( ( void * ) pSubscriptionList, 0x00, sizeof( pSubscriptionList ) );
+
+    /* This demo subscribes and publishes to topics at Qos1, so the publish
+     * messages received from the broker should have QoS1 as well. */
+    pSubscriptionList[ 0 ].qos = MQTTQoS1;
+    pSubscriptionList[ 0 ].pTopicFilter = pTopicFilter;
+    pSubscriptionList[ 0 ].topicFilterLength = topicFilterLength;
+
+    /* Generate packet identifier for the SUBSCRIBE packet. */
+    lastSubscribePacketIdentifier = MQTT_GetPacketId( pMqttContext );
+
+    /* Send SUBSCRIBE packet. */
+    mqttStatus = MQTT_Subscribe( pMqttContext,
+                                 pSubscriptionList,
+                                 sizeof( pSubscriptionList ) / sizeof( MQTTSubscribeInfo_t ),
+                                 lastSubscribePacketIdentifier );
+
+    if( mqttStatus != MQTTSuccess )
+    {
+        LogError( ( "Failed to send SUBSCRIBE packet to broker with error = %u.",
+                    mqttStatus ) );
+        returnStatus = EXIT_FAILURE;
+    }
+    else
+    {
+        LogInfo( ( "SUBSCRIBE sent for topic %.*s to broker.\n\n",
+                   topicFilterLength,
+                   pTopicFilter ) );
+
+        /* Wait for acknowledgement packet (SUBACK) from the broker in response to the
+         * subscribe request. */
+        mqttStatus = MQTT_ProcessLoop( pMqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+
```

Comment:
If the intention here is to make sure the SUBACK was received, we could add a flag that's set in the event callback on SUBACK reception. Just a thought, I don't expect it to be changed for this PR.

Reply:
If the suggestion is to call `ProcessLoop` avoid the large timeout value, we can consider that for a separate PR effort.